src/main/java/net/fabricmc/nameproposal/field/FieldData.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2023 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.nameproposal.field;
+
+public record FieldData(
```

Comment:
I wonder if it would be better to sotore the method/field node here, and then maybe have methods to get the field/method name from there?

Reply:
I think that would work. My main concern was that it might not cover every type of field initialization beyond the ones currently supported, but that issue can be ignored if breakage is fine.